### ISSUE TITLE
Windows clickonce installer

### ISSUE THREAD

**Issue by athairus** [team-phoenix/Phoenix#119]
Install using clickonce. Apps that use it include: Google Chrome and Github for Windows.

Gotta find something similar for OS X. Linux users are on your own... maybe a PPA for Debian derivatives?

**Comment by RonnChyran** [comment]
ClickOnce is a terrible way to distribute software. It doesn't allow the user to choose the install path and installs it in an extremely obscure directory inside the user AppData, and relies on a server to host the install files, which are packaged in a non-standard, Microsoft-only format, making it unable to be mirrored. If the server goes down, the ClickOnce manifest no longer works. It's useful if an offline install option is provided as well, but if ClickOnce is the only installation option, it's not ideal.

If a user-friendly web (thin-client latest version) install option is desired, what I like to do is write a custom installer that pulls a zip file of the installation files and dependencies, and wrap it up in a nice thin UI. Or you could have it download a conventional install package (Inno Setup or .msi for Windows), and run that. Windows has a package manager called chocolatey, although its usually used by programmers and power users so I wouldn't suggest that being the sole option of publishing.

On OSX, the usual method of distribution is a dmg file containing a .app folder for fully packaged applications, or a .pkg installer for apps that aren't fully contained within the .app folder and require additional setup. You could have the .app run a bootstrap that downloads the latest version as well and install it locally. Homebrew, like chocolatey, is also an option on OSX, for my technically minded folks.

And of course on Linux, PPA is the conventional install option. On Ubuntu, I suggest publishing to the Ubuntu software centre.

**Comment by athairus** [comment]
Okay, you bring up some good points against ClickOnce. I'll address them one by one.

First off, about the install path, I assumed most users that'll end up using Phoenix won't care what directory they install it to. They would just want Phoenix to be installed wherever the default is. The fact that it's installed in an 'obscure directory in AppData' isn't a bad thing, as that folder is writable without admin privileges!

About the files being hosted... I anticipate we'll have some reliable CDN for serving the install files, whose format isn't terribly important to anyone but us, the programmers. The install files being a MS-proprietary format doesn't matter terribly much if we're considering the install process on Windows only, does it? :)

Now, I'd prefer to use some pre-existing solution to do what I want to do (a dead-simple and good looking installer with auto updating), and I've only did the tiniest amount of research on ClickOnce before making this ticket. As I read about some of the hassles involved, I now realize that I might want to look into alternatives. Here's one I've found:

https://github.com/Squirrel/Squirrel.Windows

Now, with all that said, I don't intend to have this be the ONLY way to install Phoenix. I think a .msi would be good, along with a portable .zip edition.
